src/web/app/components/question-types/question-edit-answer-form/constsum-options-question-edit-answer-form.component.html

```diff
@@ -4,7 +4,7 @@
   </div>
   <div class="col-12 col-sm-5">
     <input type="number" class="form-control"
-           [ngModel]="i < responseDetails.answers.length ? responseDetails.answers[i] : ''"
+           [ngModel]="responseDetails.answers[i] || ''"
```

Comment:
It is padded with zeroes due to the submission parameters requiring 0s in the model. Otherwise half-filled array would fail the model submission when the user clicks have. I couldn't find any way to intercept the process between the user clicking save and actually submitting to server to change the value right before the actual submission. I could change the backend to accept null values though.

Reply:
The 0's is not padded. It is placeholder to enable you update the second/thrid/... option.

Reply:
@xpdavid so what do you suggest we do here? Or is the fix proposed here sufficient?

Reply:
I think I can directly give the fix to solve the problem :P
https://github.com/TEAMMATES/teammates/blob/dd9977ec150a49c93040004c4eb034a43fe0180a/src/web/app/components/question-types/question-edit-answer-form/constsum-options-question-edit-answer-form.component.ts#L38-L41

Change `0` to `RANK_RECIPIENTS_ANSWER_NOT_SUBMITTED` constant.

Keep the current logic, add a check for `RANK_RECIPIENTS_ANSWER_NOT_SUBMITTED` and display it as empty string also.

Reply:
That variable is for rank recipients question though, although the purpose do otherwise fit

Reply:
> Keep the current logic, add a check for `RANK_RECIPIENTS_ANSWER_NOT_SUBMITTED` and display it as empty string also.

I don't understand how this is better. Where would this check happen? Similar to `[ngModel]="responseDetails.answers[i] || ''"` except that it is now `[ngModel]="responseDetails.answers[i] === RANK_RECIPIENTS_ANSWER_NOT_SUBMITTED ? responseDetails.answers[i] : ''"`?

Reply:
Sorry for the confusion. There are several things here.
- `0` is a valid answer so it should not be displayed as `''` (I might be wrong of this)
- Therefore, we should only display `*_NOT_SUBMITTED` as empty string for input box
- `RANK_CONSTANT_ANSWER_NOT_SUBMITTED` is not zero and is a special value (try search it in the code base).

Reply:
@xpdavid I hate to be the bearer of bad news, but the back-end doesn't have any special treatment for that kind of value for constant sum question type. It does have for rank question, but this is not what the issue at hand is.

Reply:
Understood. However, if we want to achieve the behavior of having empty string displayed when user does not touch the box, we will need a special value here.

I think it needs more discussion on how do we want to achieve the behavior. Can we remove the commit and continue with other commits on this PR.